src/components/Room/useRoom.ts

```diff
@@ -114,8 +114,14 @@ export function useRoom(
   const { getDisplayUsername } = usePeerNameDisplay()
 
   const fileTransferService = useMemo(
-    () => new FileTransferService(roomConfig.rtcConfig!),
-    [roomConfig.rtcConfig]
+    () =>
+      new FileTransferService({
+        rtcConfig: roomConfig.rtcConfig!,
+        peerRoom,
+        useDirectFileTransferNetworking:
+          !!roomConfig.useDirectFileTransferNetworking,
```

Comment:
This should be reading the `VITE_USE_DIRECT_FILE_TRANSFER_NETWORKING` environment variable directly.